### PR TITLE
google: create virtual machines

### PR DESCRIPTION
**Pull request by jstuever**
This change adds the terraform bits to create the bootstrap and control
plane virtual machines for the google platform. It also creates groups
for use by the load balancers with one for bootstrap and one for each
control plane zone.

### CODE REVIEW

File: data/data/gcp/bootstrap/main.tf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  network_interface {
+    subnetwork = var.subnet
+  }
+
+  metadata = {
+    user-data = var.ignition
+  }
+
+  tags = ["${var.cluster_id}-control"]
```
Comment (abhinavdahiya) [inline data/data/gcp/bootstrap/main.tf]: `clusterid-master`

File: data/data/gcp/bootstrap/main.tf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  labels = var.labels
+}
+
+resource "google_compute_instance_group" "bootstrap" {
```
Comment (abhinavdahiya) [inline data/data/gcp/bootstrap/main.tf]: are instance groups a requirement?

File: data/data/gcp/bootstrap/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+output "bootstrap_instance" {
```
Comment (abhinavdahiya) [inline data/data/gcp/bootstrap/outputs.tf]: the bootstrap module is deleted by the installer after bootstrapping completes, so any resource that uses these outputs will also be marked for deletion, therefore it's best if the outputs for bootstrap module is empty.

File: data/data/gcp/bootstrap/variables.tf [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+variable "subnet" {
+  type        = string
+  description = "The subnetwork the bootstrap node will be added to."
+}
+
+variable "root_volume_size" {
+  type        = string
+  description = "The volume size (in gibibytes) for the bootstrap node's root volume."
+  default     = "30"
```
Comment (abhinavdahiya) [inline data/data/gcp/bootstrap/variables.tf]: so we use 128 for all other platforms.

File: data/data/gcp/control/main.tf [hunk 5]
```diff
@@ -0,0 +1,40 @@
+resource "google_compute_instance" "control" {
+  name         = "${var.cluster_id}-control-${count.index}"
```
Comment (abhinavdahiya) [inline data/data/gcp/control/main.tf]: this name must match the machine object's name for adoption... 
https://github.com/openshift/installer/blob/21570f8a4ab99a9de874397e6fbe42ba55dbd29a/pkg/asset/machines/gcp/machines.go#L48

``clusterid-master-idx`

File: data/data/gcp/control/main.tf [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  count        = var.instance_count
```
Comment (abhinavdahiya) [inline data/data/gcp/control/main.tf]: usually count is first field and better to separate from the rest with a newline...

File: data/data/gcp/control/main.tf [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  network_interface {
+    subnetwork = var.subnet
+  }
+
+  metadata = {
+    user-data = var.ignition
+  }
+
+  tags = ["${var.cluster_id}-control"]
```
Comment (abhinavdahiya) [inline data/data/gcp/control/main.tf]: same as https://github.com/openshift/installer/pull/1940/files#r300070723

File: data/data/gcp/control/main.tf [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+  labels = var.labels
+}
+
+resource "google_compute_instance_group" "control" {
```
Comment (abhinavdahiya) [inline data/data/gcp/control/main.tf]: same as https://github.com/openshift/installer/pull/1940/files#r300072322

File: data/data/gcp/control/variables.tf [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+variable "subnet" {
+  type        = string
+  description = "The subnetwork the control plane nodes will be added to."
+}
+
+variable "root_volume_size" {
+  type        = string
+  description = "The size of the volume in gigabytes for the root block device."
+  default     = "30"
```
Comment (abhinavdahiya) [inline data/data/gcp/control/variables.tf]: `128`

File: data/data/gcp/network/firewall-compute.tf [hunk 10]
```diff
@@ -7,7 +7,7 @@ resource "google_compute_firewall" "compute_ingress_icmp" {
   }
 
   source_ranges = [var.network_cidr]
-  target_tags   = ["${var.cluster_id}-worker"]
+  target_tags   = ["${var.cluster_id}-compute"]
```
Comment (abhinavdahiya) [inline data/data/gcp/network/firewall-compute.tf]: https://github.com/openshift/installer/pull/1938/commits/c11f2d2a72f3016842c9e50f2a866a10018d1f3b

needs to stay master and worker

File: data/data/gcp/variables-google.tf [hunk 11]
```diff
@@ -1,9 +1,36 @@
-variable "google_project" {
```
Comment (abhinavdahiya) [inline data/data/gcp/variables-google.tf]: https://github.com/openshift/installer/pull/1938/commits/388c54599c9c4995a521225a7064f2698475ce04

for making sure the prefix is gcp not google

File: data/data/gcp/bootstrap/main.tf (hunk 2 above)
Comment (jstuever) [reply data/data/gcp/bootstrap/main.tf]: @abhinavdahiya I'm not sure, we can punt this to CORS-1120

**Pull request merged**